packages/noya-state/src/selectors/layerSelectors.ts

```diff
@@ -501,3 +501,15 @@ export function getConstrainedScaling(
       true)
   );
 }
+
+// Check if layer is visible in the page
+export function isLayerVisible(layer: Sketch.AnyLayer, canvasFrame: Rect) {
```

Comment:
I think we may want to use `getLayersInRect`, which will also handle transforms

https://github.com/noya-app/noya/blob/e3c896a128675123ec46608740f38d5f98767367/packages/noya-state/src/selectors/geometrySelectors.ts#L443-L452